recipes/server_check.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require_relative './server_check_helper'
+
+consul_service_folder = '/etc/systemd/system/consul.service.d'
+drop_in_file = "#{consul_service_folder}/server_start_prevention.conf" # prevent systemd from starting
+
+directory consul_service_folder do
+  owner 'root'
+  mode '0755'
+  action :create
+  recursive true
+  not_if { ::File.directory?(consul_service_folder) }
+end
+
+# Output file is used for systemd to check whether it should start consul.
+if is_server_allowed?(node['consul']['config']['retry_join'], node['ipaddress'])
+  # For non rogue servers, simply remove the drop-in file to unblock service from starting.
+  if File.exist?(drop_in_file)
+    file drop_in_file do
+      action :delete
+    end
+
+    reload_systemd()
+  end
+else
+  # Currently has to use ConditionPathExists instead of ExecCondition.
+  # Blocks the service from starting if this file itself is present.
+  # ExecCondition requires systemd 243+, Ubuntu 1804 has systemd 237.
+  # There is an alternative to use ExecStartPre to block service from starting,
+  # however ExecStartPre on its own would need to exit on a non 0 code in order
+  # to not start the service, this would fail the entire chef run.
+  # Another alternative is to combine ExecStartPre and ConditionPathExists by
+  # moving all the logic into ExecStartPre script, however currently we might
+  # prefer to keep the logic in the chef ecosystem.
+  file drop_in_file do
+    owner 'root'
+    mode '0644'
+    content <<-EOF.gsub(/^ */, '')
+        [Unit]
+        ConditionPathExists=!#{drop_in_file}
```

Comment:
It still feels weird to me that we are checking the drop-in file for this check, but I don't think it's wrong. I would personally prefer to delegate this logic to a script and rely on `ExecStartPre` and a status code. However, I understand this adds more complexity and we may want to go with this already tested and validated approach.